### PR TITLE
Test generated_plugin_registrant analysis & suppress long lines in web plugin registrant

### LINKED ISSUE #71980: [web] Plugin registrant generates lines longer than 80 chars
## Description

This behavior affects the `beta`+ channels with _web support enabled_.

Whenever running `pub get` and having the [`lines_longer_than_80_chars` lint](https://dart-lang.github.io/linter/lints/lines_longer_than_80_chars.html) enabled, the analyzer will complain about lines being longer than 80 chars in `generated_plugin_registrant.dart`.

image

### PR DESCRIPTION
**Pull request by creativecreatorormaybenot**
## Description

This PR suppresses [`lines_longer_than_80_chars`](https://dart-lang.github.io/linter/lints/lines_longer_than_80_chars.html) in `generated_plugin_registrant.dart`.

This is a real annoyance at this moment:

image

The warnings appear every time `pub get` is run.

### The lint

I believe that this PR is justified because the lint ([docs](https://dart-lang.github.io/linter/lints/lines_longer_than_80_chars.html)) is part of the **Effective Dart** style guide.

## Related Issues

* Resolves #71980.

## Tests

Not sure how to test for lints.

## Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.

### CODE REVIEW

**Comment by Hixie** [comment]
Does the failure only occur when there's a particular lint set in the app? Maybe one way to test this would be to have a test that runs `flutter create`, then adds a plugin with a particularly long name, then enables the relevant lint, then runs `flutter pub get` and `flutter analyze` and checks there's no errors. that would be a good thing to check anyway, it would catch other regressions too. We may even already have a test you can just slightly tweak to do this. Maybe check the tests in `packages/flutter_tools/test`, or in `dev/`.

**Comment by creativecreatorormaybenot** [comment]
@Hixie I gave it a look. 

To be honest, I was not really able to find much as a starting point. Is that whole testing stuff documented somewhere?

It seems like it could be in any of these:

* Something like `/dev/.../analyze_test.dart`: https://github.com/flutter/flutter/blob/5cfb16b193886e68e51cc7256389d78a21f31bfd/dev/bots/test/analyze_test.dart - here I am not sure how the test is run in general.
* `flutter_tools/test/commands.shard/` seems to contain tests that go in this direction, but I would not be testing a command in this case.
* Would my test go into `general.shard`?
* Or into `web.shard`?

Frankly, it all seems very involved and I was not able to make good progress / find an actual starting point - only looked into a lot of files and found bits and pieces that would fit; I am sure I missed something obvious - it would just require a little more time / guidance for more efficiency.

**Comment by creativecreatorormaybenot** [comment]
@Hixie I found `_createAndAnalyzeProject` in a `commands.shard` test file: https://github.com/flutter/flutter/blob/02eab12fd3e2ae35e8dc51dc046d556629e6d1be/packages/flutter_tools/test/commands.shard/permeable/create_test.dart#L2373

However, I am still struggling to put the pieces together:

* My test would not be a command test, so I would have to move that code out of there somehow.
* The whole function is undocumented, so I am generally lost as to how to use it.
* Where would I place my tests?

**Comment by Hixie** [comment]
It could be a command test. It could live in create_test.dart in fact, if you want it to. You could also just copy that file and make a new one in one of the other directories and start from that.
To run the tests you do `../../bin/cache/dart-sdk/bin/pub run test test/general.shard` or similar (depends what shard your test is in), from the packages/flutter_tool directory.
https://github.com/flutter/flutter/tree/master/packages/flutter_tools/test/integration.shard is probably the right directory.

Please create a README.md file wherever it is you looked for the documentation on how these tests run, and add anything you learn there, so that the next person can learn these same things faster. :-)

**Comment by creativecreatorormaybenot** [comment]
@Hixie I actually found out that the syntax is not the same as in the screenshot anymore (changed from `Plugin.registerWith(registry.registrarFor(Plugin))` to `Plugin.registerWith(registrar)`), which makes it more unlikely to exceed 80 chars.

However, I did add the regression test and I think that this PR is still useful because it tests that the analysis passes in general.

I also confirmed that without my `ignore_for_file`, the regression test fails :)

**Comment by creativecreatorormaybenot** [comment]
😄 @Hixie so it seems that this was actually your adjustment from just a few days before I created my PR (#70337).

File: packages/flutter_tools/test/integration.shard/README.md
Comment (Hixie) [inline packages/flutter_tools/test/integration.shard/README.md:14]: the tool's name is lowercase (but if you like you can put it in backticks, as in "the \`flutter\` tool") to make it clearer)

**Comment by Hixie** [comment]
This looks great!

File: packages/flutter_tools/test/integration.shard/generated_plugin_registrant_test.dart
Comment (Hixie) [inline packages/flutter_tools/test/integration.shard/generated_plugin_registrant_test.dart:49]: formatting here is a bit weird (looks like `expect` ends with the `}`). If you are using `dartfmt` then you can probably get a better formatting by putting a comma after `exists`.

File: packages/flutter_tools/test/integration.shard/generated_plugin_registrant_test.dart
Comment (Hixie) [inline packages/flutter_tools/test/integration.shard/generated_plugin_registrant_test.dart:85]: ditto

File: packages/flutter_tools/test/integration.shard/generated_plugin_registrant_test.dart
Comment (Hixie) [inline packages/flutter_tools/test/integration.shard/generated_plugin_registrant_test.dart:80]: prefer putting the `\n` at the end of the previous line rather than the start of the new line

Comment (creativecreatorormaybenot) [reply packages/flutter_tools/test/integration.shard/generated_plugin_registrant_test.dart:80]: The problem here is that I wanted to separate the package key and the version value - I will refactor this to allow for `path` properly.

File: packages/flutter_tools/test/integration.shard/generated_plugin_registrant_test.dart
Comment (Hixie) [inline packages/flutter_tools/test/integration.shard/generated_plugin_registrant_test.dart:104]: formatting is confusing here. Looks like the '..' is less nested than the previous line but it's actually more nested?

If you're using `dartfmt` you can probably get a better effect by putting a comma between the two `))`s at the end of the statement (line 109 below).

File: packages/flutter_tools/test/integration.shard/generated_plugin_registrant_test.dart
Comment (Hixie) [inline packages/flutter_tools/test/integration.shard/generated_plugin_registrant_test.dart:116]: this seems to just create a file and then leave it there?

File: packages/flutter_tools/test/integration.shard/README.md
Comment (creativecreatorormaybenot) [reply packages/flutter_tools/test/integration.shard/README.md]: Thanks! That confused me - I also added a section on how to download the Dart SDK into the path.

File: packages/flutter_tools/test/integration.shard/generated_plugin_registrant_test.dart
Comment (Hixie) [inline packages/flutter_tools/test/integration.shard/generated_plugin_registrant_test.dart:158]: really trivial nit: lots of blank lines here that aren't necessary

(we use blank lines to separate significant parts of the code)

File: packages/flutter_tools/test/integration.shard/generated_plugin_registrant_test.dart
Comment (Hixie) [inline packages/flutter_tools/test/integration.shard/generated_plugin_registrant_test.dart:187]: similar problem to earlier

File: packages/flutter_tools/test/integration.shard/generated_plugin_registrant_test.dart
Comment (creativecreatorormaybenot) [reply packages/flutter_tools/test/integration.shard/generated_plugin_registrant_test.dart:116]: I am actually not sure about this 🤔 

I found this in `create_test` and I included it here because I also used the analysis function, which seems to depend on `flutter_tools.snapshot`. Here is the origin: https://github.com/flutter/flutter/pull/39338

**Comment by creativecreatorormaybenot** [comment]
@Hixie Everything should be addressed 👍 

I resolved the trivial suggestions that I did not respond to. Do you prefer me resolving trivial ones or leaving all of them open for you to resolve?

File: packages/flutter_tools/test/integration.shard/generated_plugin_registrant_test.dart
Comment (Hixie) [reply packages/flutter_tools/test/integration.shard/generated_plugin_registrant_test.dart:123]: cc @zanderso

Comment (zanderso) [reply packages/flutter_tools/test/integration.shard/generated_plugin_registrant_test.dart:123]: See https://github.com/flutter/flutter/blob/master/packages/flutter_tools/test/commands.shard/permeable/create_test.dart#L73

Comment (creativecreatorormaybenot) [reply packages/flutter_tools/test/integration.shard/generated_plugin_registrant_test.dart:123]: @zanderso Oh, I see! I will add that part 👍

**Comment by creativecreatorormaybenot** [comment]
@Hixie @zanderso I added restoring the tools snapshot 👌 

Thanks for pointing it out!

**Comment by Hixie** [comment]
This looks great to me, but I'd feel more comfortable getting a second review from someone closer to this code. Maybe @christopherfujino or @zanderso?

**Review by zanderso — APPROVED** [review]

**Pull request merged**
